### ISSUE TITLE
Collapsible fields lack accessibility

### ISSUE THREAD

**Issue by wbobeirne** [fedimint/ui#324]
The collapsible field group headings do not allow for keyboard highlighting or opening / closing without using the mouse. They should use a `<button>` element or have tabIndex / aria fields to allow for keyboard interaction. They should also have titles for screen readers that indicate what interacting with them do.

![CleanShot 2023-11-28 at 09 58 14@2x](https://github.com/fedimint/ui/assets/649992/a4b0cf0f-0bc8-464d-89a3-c3d84c0b57fd)
